### PR TITLE
test: scrollInputIntoView method coverage in all pages

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at e800a48</samp>

Added `getActiveElement` function and tests for scrolling input into view on keydown events in various pages. Removed unnecessary commas to follow code style guide.

### <samp>🤖 Generated by Copilot at e800a48</samp>

> _Sing, O Muse, of the code that was changed by the skillful coder_
> _Who added new functions to test the `scrollInputIntoView` feature_
> _And removed the commas that cluttered the arguments of many a function_
> _Making the code more elegant and pleasing to the eye of the reader_

### <samp>🤖 Generated by Copilot at e800a48</samp>

*  Remove extra commas after the last arguments of various functions and operators to follow the code style guide ([link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L147-R151), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6R218-R237), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L589-R609), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L618-R638), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L648-R668), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L678-R698), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L685-R705), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L715-R735), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L728-R748), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L758-R778), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L787-R807), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L793-R813), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L818-R838), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L844-R864), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L850-R870), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L874-R894), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L880-R900), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L975-R995), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1004-R1024), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1028-R1048), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1050-R1070), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1061-R1081), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1081-R1101), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1092-R1112), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1112-R1132), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1161-R1181), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL284-R284), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933L507-R527), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933L517-R537), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933L532-R552), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933L550-R570), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933L562-R582), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933L578-R598), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL447-R451), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL473-R477), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL490-R497), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL508-R512), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL525-R534), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL572-R576), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL583-R590), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL598-R602), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL610-R614), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL619-R623), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL649-R653), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL671-R678), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL700-R704), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL714-R720), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL733-R739), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL745-R749), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL767-R778), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL811-R815), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL827-R831), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL897-R906), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL919-R923), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL928-R936), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL939-R944), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL947-R952), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL958-R964), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL973-R977), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL984-R988), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1003-R1012), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1016-R1024), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1044-R1050), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1054-R1059), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1080-R1086), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1112-R1120), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1122-R1126), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1139-R1143), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1145-R1149), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1151-R1155), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1184-R1188), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1190-R1194), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1199-R1203), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1211-R1215), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1221-R1226), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1229-R1238), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1248-R1258), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1270-R1274), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1280-R1284), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1295-R1300), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1309-R1314), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1318-R1326), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1333-R1342), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1358-R1362), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1376-R1380), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1384-R1394), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1413-R1420), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1445-R1449), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1558-R1562), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1566-R1570), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1623-R1627), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1637-R1641), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1657-R1661), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1696-R1700), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1731-R1735), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1750-R1759), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1801-R1811), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1827-R1831), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1862-R1870), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1872-R1876), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1886-R1890), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1909-R1913), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1927-R1936), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1976-R1980), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL1997-R2001), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2033-R2037), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2041-R2045), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2048-R2052), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2062-R2066), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2074-R2084), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2086-R2090), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2240-R2244), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2293-R2298))
* Add `getActiveElement` function to `AddEditExpensePage`, `AddEditMileagePage`, and `AddEditPerDiemPage` classes to return the active element in the DOM and make the code more testable and modular ([link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR486-R489), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eR357-R360), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dR319-R322))
* Use `getActiveElement` function instead of directly accessing the `document.activeElement` property in `AddEditExpensePage`, `AddEditMileagePage`, and `AddEditPerDiemPage` classes ([link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL478-R478), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL349-R349), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL311-R311))
* Add `scrollInputIntoView` and `getActiveElement` functions to the `TestCases1` function in `add-edit-expense-1.spec.ts`, `add-edit-mileage-1.spec.ts`, and `add-edit-per-diem-1.page.spec.ts` files to test the new functionality of scrolling the input element into view on keydown events ([link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6R218-R237), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961R267-R286), [link](https://github.com/fylein/fyle-mobile-app/pull/2484/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933R164-R183))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-mileage/add-edit-mileage-1.spec.ts [hunk 1]
```diff
@@ -264,6 +264,26 @@ export function TestCases1(getTestBed) {
       expect(component.isExpandedView).toBeFalse();
     });
 
+    it('scrollInputIntoView(): should scroll input into view on keydown', () => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-1.spec.ts:267]: this is newly added method

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts [hunk 2]
```diff
@@ -161,6 +161,26 @@ export function TestCases1(getTestBed) {
       expect(component).toBeTruthy();
     });
 
+    it('scrollInputIntoView(): should scroll input into view on keydown', () => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts:164]: this is newly added test

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 3]
```diff
@@ -308,14 +308,18 @@ export class AddEditPerDiemPage implements OnInit {
 
   @HostListener('keydown')
   scrollInputIntoView(): void {
-    const el = document.activeElement;
+    const el = this.getActiveElement();
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:311]: changes

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 4]
```diff
@@ -308,14 +308,18 @@ export class AddEditPerDiemPage implements OnInit {
...
     if (el && el instanceof HTMLInputElement) {
       el.scrollIntoView({
         block: 'center',
       });
     }
   }
 
+  getActiveElement(): Element {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:319]: changes

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 5]
```diff
@@ -215,6 +215,26 @@ export function TestCases1(getTestBed) {
       expect(component).toBeTruthy();
     });
 
+    it('scrollInputIntoView(): should scroll input into view on keydown', () => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:218]: newly added tests

**Comment by suyashpatil78** [comment]
Rest are all prettier changes

**Review by OmkarJ13 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**